cmd/dendrite-monolith-server/main.go

```diff
@@ -67,9 +67,15 @@ func main() {
 
 	accountDB := base.CreateAccountsDB()
 	deviceDB := base.CreateDeviceDB()
-	keyDB := base.CreateKeyDB()
 	federation := base.CreateFederationClient()
-	keyRing := keydb.CreateKeyRing(federation.Client, keyDB, cfg.Matrix.KeyPerspectives)
+
+	serverKeyAPI := serverkeyapi.SetupServerKeyAPIComponent(
+		base, federation,
+	)
+	if base.EnableHTTPAPIs {
+		serverKeyAPI = base.CreateHTTPServerKeyAPIs()
```

Comment:
API wise I'd like to combine these into a single access point per internal API. At the moment, I never know if I should be using `base.CreateFoo` or `SetupFooAPI` etc. I appreciate we can't have one **function** since the HTTP form has no need to dependency inject but the actual concrete API impl does need to inject stuff.

If we just had:
```go
api.NewFooHTTP() *Foo
api.NewFoo(a,b,c Deps) *Foo
```
that would make this a lot nicer I think.

What are your thoughts? If we do want to do this, I'd propose we add it to the board.

Reply:
I definitely think we can improve the naming for all of this for all components, yes.